### PR TITLE
Plasmamen don't get damaged from slurping plasma anymore

### PR DESCRIPTION
**Pull request by Irradiation**
I guess current plasmemes are fine with gaseous plasma keeping them alive but god forbid they ingest LIQUID plasma, because that shit ain't kosher.

[qol][muh immulsions]

:cl:
 * tweak: Plasmamen don't take toxic damage from ingesting plasma anymore.

### CODE REVIEW

**Comment by Gringonius** [comment]
It should actually heal them too if you ask me. I'd make it work like DD but I'm not sure about the balance issues that could actually cause.

**Comment by otatoh** [comment]
[the plasmaman metaclub with remember this]

**Comment by Kurfursten** [comment]
@Gringonius >Be scientist/mechanic/miner
>Carry one stack of blasma and a mortar
>Have 1k DD-equivalent onhand at all times

**Comment by SGuivenne** [comment]
@Kurfursten lets be honest, in most lethal combat you don't die to bleedout or long term wounds anyway, and with the space suit exposure deaths are off the table too, what kills a plasmaman and most crew will be when they get shot in the head 8 times until their brain pops out or get eviserated by a traitor with an energy sword.  

This doesn't give them anything that a meta player couldn't already do.

**Comment by Kurfursten** [comment]
It makes it a lot easier to carry around a fuckton of DD. In practice nobody mixes up 1,000 Doctor's Delight because it's a pain to make compared to other chems but the 5u autoinjectors of DD get used every round.

**Comment by bathosbathtime** [comment]
I would have it weaker than tricord or work like nitrogen to vox if it heals at all.  Just a fun interaction.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -2288,10 +2288,14 @@
 
 	if(..())
 		return 1
-
+		
+	if(M.species.name=="Plasmaman")
+		return
+	else
+		M.adjustToxLoss(3 * REM)
```
Comment (SonixApache) [inline code/modules/reagents/Chemistry-Reagents.dm:2296]: -3, come on, you *just* made the same mistake in the vox one
Did you even test this?

Comment (Irradiation) [reply code/modules/reagents/Chemistry-Reagents.dm]: No, I was at college and intended to fix it when I got home.

Comment (ShiftyRail) [reply code/modules/reagents/Chemistry-Reagents.dm:2296]: Wait
This for *normal* people
So it would indeed be a +3

**Comment by Dacendeth** [comment]
I was going to link you an issue I made related to this that this pr would close but it turns out it already got closed for "not a bug" 😅

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 1 above)
Comment (SonixApache) [reply code/modules/reagents/Chemistry-Reagents.dm:2296]: And here I thought plasmemes would get *healed* by plasma.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Plasmamen being healed by plasma is like one of those nice little easter eggs that would absolutely get abused to hell and back by one powergamer

File: code/modules/reagents/Chemistry-Reagents.dm (hunk 1 above)
Comment (Irradiation) [reply code/modules/reagents/Chemistry-Reagents.dm]: >/datum/reagent/plasma/on_mob_life(var/mob/living/M)
	if(..())
		return 1
	if(holder.has_reagent("inaprovaline"))
		holder.remove_reagent("inaprovaline", 2 * REM)
	M.adjustToxLoss(3 * REM)


I haven't touched the effects of Plasma on people.

Comment (SonixApache) [reply code/modules/reagents/Chemistry-Reagents.dm:2296]: yeah i'm retarded ignore all this

**Comment by spoonylove** [comment]
Somebody make a bar drink, thats simpler to make than DD, with plasma for the plasbros, and have that work like DD, that way they have to take SOME steps to get the healing benefits. Call it like RATTLIN BONES or some shit.

**Comment by Kurfursten** [comment]
Degenerate Calcium + Plasma
Calls for milk and mutagen, not as involved as tricord (inaprov+antitox)+4 juices but still a reasonable roadblock involving both chemistry and service. Plus, calcium.

**Comment by Dacendeth** [comment]
It's actually really easy to carry 1000+ units of DD on you now with cauldrons and chemmachine slurping exist, just need to get the cartons of juice/milk.

**Comment by Irradiation** [comment]
I mean, if somebody makes a drink sprite, I'm sure that can be arranged, but that's outside the purpose of this PR.

**Comment by Irradiation** [comment]
Problem fixed and tested. It werks now.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -2288,10 +2288,15 @@
 
 	if(..())
 		return 1
-
+		
+	var/mob/living/carbon/human/H = M
+	if(H.species.name=="Plasmaman")
```
Comment (ShiftyRail) [inline code/modules/reagents/Chemistry-Reagents.dm]: Replace that with `isplasmaman(H)` to avoid eventual runtimes

**Pull request merged**